Source/WebCore/dom/SimpleRange.cpp

```diff
@@ -102,6 +114,62 @@ IntersectingNodeIterator::IntersectingNodeIterator(const SimpleRange& range)
     enforceEndInvariant();
 }
 
+static RefPtr<Node> firstIntersectingNode(const SimpleRangeWeak& range)
+{
+    if (!range.start.container)
+        return nullptr;
+    if (range.start.container->isCharacterDataNode())
+        return range.start.container.get();
+    if (auto child = range.start.container->traverseToChildAt(range.start.offset))
+        return child;
+    return NodeTraversal::nextSkippingChildren(*(range.start.container));
+}
+
+static RefPtr<Node> nodePastLastIntersectingNode(const SimpleRangeWeak& range)
+{
+    if (!range.end.container)
+        return nullptr;
+    if (range.end.container->isCharacterDataNode())
+        return NodeTraversal::nextSkippingChildren(*(range.end.container));
+    if (auto child = range.end.container->traverseToChildAt(range.end.offset))
+        return child;
+    return NodeTraversal::nextSkippingChildren(*(range.end.container));
+}
+
+static RefPtr<Node> firstIntersectingNodeWithDeprecatedZeroOffsetStartQuirk(const SimpleRangeWeak& range)
+{
+    if (!range.start.container)
+        return nullptr;
+    if (range.start.container->isCharacterDataNode())
+        return range.start.container.get();
+    if (auto child = range.start.container->traverseToChildAt(range.start.offset))
+        return child;
+    if (!range.start.offset)
+        return range.start.container.get();
+    return NodeTraversal::nextSkippingChildren(*(range.start.container));
+}
+
+template<TreeType treeType> Node* commonInclusiveAncestor(const SimpleRangeWeak& range)
+{
+    if (!range.start.container || !range.end.container)
+        return nullptr;
+    return commonInclusiveAncestor<treeType>(*(range.start.container), *(range.end.container));
+}
+
+IntersectingNodeIterator::IntersectingNodeIterator(const SimpleRangeWeak& range)
+    : m_node(firstIntersectingNode(range))
+    , m_pastLastNode(nodePastLastIntersectingNode(range))
+{
+    enforceEndInvariant();
+}
+
+IntersectingNodeIterator::IntersectingNodeIterator(const SimpleRangeWeak& range, QuirkFlag)
+    : m_node(firstIntersectingNodeWithDeprecatedZeroOffsetStartQuirk(range))
+    , m_pastLastNode(nodePastLastIntersectingNode(range))
+{
+    enforceEndInvariant();
+}
+
```

Comment:
For SimpleRangeWeak I overloaded these functions that were provided by SimpleRange. They are not currently being used though by whoever needs to access  the members m_dragStartSelection. Should these just be deleted because they are not currently needed? Or, should I define all the equivalent methods that SimpleRange implements to operate on SimpleRanges so the two can have similar usages if ever needed?